arch/riscv/core/isr.S

```diff
@@ -557,6 +557,7 @@ irq_done:
 
 check_reschedule:
 
+#ifdef CONFIG_MULTITHREADING
```

Comment:
nit:
```
#ifndef CONFIG_MULTITHREADING
    j no_reschedule
#endif
```

Reply:
`z_get_next_switch_handle` is not defined if MULTITHREADING is disabled.
This conditional branching needs.

Reply:
right